### PR TITLE
Purge sdi polish

### PR DESCRIPTION
**Pull request by PietroPasotti**
## Issue
Lib was missing some typing and docstrings, had complex type signatures and redundant code paths.

## Solution
A bit of polishing.

## Testing Instructions
No (production) code paths should have been altered. Utests/itests should be enough.

## Release Notes
Added types and return type annotations throughout.

### CODE REVIEW

**Review by jnsgruk — APPROVED** [review]

File: lib/charms/traefik_k8s/v0/ingress_per_unit.py [hunk 1]
```diff
@@ -116,18 +116,26 @@ def _handle_ingress_per_unit(self, event):
...
+
+
+KeyValueMapping = Dict[str, str]
+RequirerUnitData = Dict[Unit, KeyValueMapping]
+ProviderApplicationData = Dict[str, KeyValueMapping]
+
+
 # ======================= #
 #  SERIALIZATION UTILS    #
 # ======================= #
```
Comment (simskij) [inline lib/charms/traefik_k8s/v0/ingress_per_unit.py:131]:
```suggestion

KeyValueMapping = Dict[str, str]
RequirerUnitData = Dict[Unit, KeyValueMapping]
ProviderApplicationData = Dict[str, KeyValueMapping]

```

I would omit these comments, or at least tone them down.

Comment (jnsgruk) [reply lib/charms/traefik_k8s/v0/ingress_per_unit.py:131]: Agree, I'm not a big fan.

Comment (PietroPasotti) [reply lib/charms/traefik_k8s/v0/ingress_per_unit.py]: done!

**Comment by jnsgruk** [comment]
LGTM - if @simskij is happy lets get it merged.

**Pull request merged**